src/viadot/orchestration/prefect/utils.py

```diff
@@ -28,6 +28,394 @@
+        Returns a list of dynamically created dates generated based on `unit` in an ascending order.
+
+        Possible values of `unit` correspond to different date formatting styles:
+            - 'years': Return a date formatted with a pendulum token 'Y'.
+            - 'months': Return a date formatted with a pendulum token 'YMM'.
+            - 'days': Return a date with a pendulum token 'YMMDD'.
+        Args:
+            number (int): The number of units from the current year to include.
+            unit (str): The unit of time ('years', 'months', 'days').
+
+        Returns:
+            List[str]: A list of dates in ascending order.
+        """
+        if unit == "years":
+            return self.generate_years(
+                last_years=number, from_year=None, num_years=None
+            )
+        elif unit == "months":
+            return self.generate_months(last_months=number)
+        elif unit == "days":
+            return self.generate_dates(last_days=number)
+
+    def handle_data_ranges(self, match: List[tuple], key: str) -> List[str]:
+        """
+        Direct execution of a specific function based on the provided value of `key`.
+        Returns a list of dynamically created dates generated based on `key`.
+
+        Depending on a unit ('years'/'months'/'days') the `match` referrs to, date formatting style differs:
+            - 'years': Return a date formatted with a pendulum token 'Y'.
+            - 'months': Return a date formatted with a pendulum token 'YMM'.
+            - 'days': Return a date with a pendulum token 'YMMDD'.
+        Args:
+            match (List[tuple]): List of every pattern match that occurs in a given string.
+            key (str): Key phrase that determines the execution of a specific function.
+        Returns:
+            List[str]: A list of dates in string format, in ascending order.
+        """
+        if key == "last_x_units":
+            for number, unit in match:
+                number = int(number)
+                return self.generate_dates_based_on_unit(number, unit)
+
+        elif key == "y_years_from_x":
+            for number, start_year in match:
+                number = int(number)
+                start_year = int(start_year)
+                replacement = self.generate_years(
+                    last_years=None, from_year=start_year, num_years=number
+                )
+                return replacement
+
+        elif key == "first_x_days_from":
+            for num_days, month_name, year in match:
+                num_days = int(num_days)
+                replacement = self.process_first_days(month_name, year, num_days)
+                return replacement
+
+        elif key == "last_x_days_from":
+            for num_days, month_name, year in match:
+                num_days = int(num_days)
+                replacement = self.process_last_days(month_name, year, num_days)
+                return replacement
+
```

Comment:
You have only if and elif here. Add `return None` or `[]` at the end to be able to return anything if key is different than all listed cases. Check what would be valid here. If return None then add `| None` to the return type.